docs/configuration/integrations/email.md

```diff
@@ -2,25 +2,36 @@
 
 Follow the instructions below to set up Codacy to send emails using your SMTP server:
 
-1.  Edit the file `values-production.yaml`, set `global.email.enabled: "true"` and define the remaining values with the credentials for your SMTP server:
+1.  Edit the file `values-production.yaml` that you used to [install Codacy](../../index.md#helm-upgrade).
+
+2.  Set `global.email.enabled: "true"` and define the remaining values with the credentials for your SMTP server:
 
     ```yaml
     email:
-        enabled: "true"
-        replyTo: "notifications@mycompany.com"
-        smtp:
-            hostname: "smtp.mycompany.com"
-            protocol: "smtps" # smtps | smtp
-            username: "my-smtp-username"
-            password: "a-s!r0ng-pVsSwORG"
-            # port: 25
+      enabled: "true"
+      replyTo: "notifications@mycompany.com" # Reply-to field on sent emails
+      smtp:
+        hostname: "smtp.mycompany.com" # Hostname of your SMTP server
+        protocol: "smtps" # SMTP protocol to use, either smtps or smtp
+        username: "my-smtp-username" # Username to authenticate on your SMTP server
+        password: "a-s!r0ng-pVsSwORG" # Password to authenticate on your SMTP server
+        port: 465 # Port of your SMTP server
```

Comment:
The idea for some fields to be commented was because they were optional.
Actually here username, password and port are optional depending on the smtp configuration of the server

Reply:
On the `values-production.yaml` only the port was commented out twice:

```
  email: {}
    # enabled: "false"
    # replyTo: "notifications@codacy.com"
    # smtp:
    #   protocol: "smtps" # smtps | smtp
    #   hostname: "smtp.codacy.com"
    #   username: "codacy"
    #   password: "a-s!r0ng-pVsSwORG"
    #   # port: 25
```

So I interpreted this as "if you omit the port, it will use port 25 by default".

But indeed, it's perfectly possible to have an SMTP server with no authentication, especially if it's internal and protected with a firewall.

Is there any difference between the two options below?

```
  email: {}
    # enabled: "false"
    # replyTo: "notifications@codacy.com"
    # smtp:
    #   protocol: "smtps" # smtps | smtp
    #   hostname: "smtp.codacy.com"
    #   # username: "codacy"
    #   # password: "a-s!r0ng-pVsSwORG"
    #   # port: 25
```

or

```
  email: {}
    # enabled: "false"
    # replyTo: "notifications@codacy.com"
    # smtp:
    #   protocol: "smtps" # smtps | smtp
    #   hostname: "smtp.codacy.com"
    #   username: ""
    #   password: ""
    #   port: ""
```

If Codacy interprets the empty values in the same way as if they are not defined, I would prefer to have the comments saying that the settings are optional, as this is more explicit:

```
  email: {}
    # enabled: "false"
    # replyTo: "notifications@codacy.com"
    # smtp:
    #   protocol: "smtps" # smtps | smtp
    #   hostname: "smtp.codacy.com"
    #   username: "" # Optional username to authenticate on your SMTP server
    #   password: "" # Optional password to authenticate on your SMTP server
    #   port: "" # Optional port of your SMTP server, the default is 25
```

Reply:
You need to comment them to make them undefined.
It might work with helm if string is empty, but I would definitely not rely on that behaviour.

Reply:
-hmmm- I see... It's better to include the "extra comment" then.